# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: 0b4595b2c153
# Full Hash: 0b4595b2c1531b83fc24fbf9b0f6ceacfc3506bd
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-22 09:35:42
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 4363e3a3d799 -r 0b4595b2c153 docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Wed Apr 22 01:37:52 2020 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Wed Apr 22 01:37:55 2020 +0000
@@ -33,14 +33,14 @@
 #define USER_ACTIVATION_LOG(msg, ...) \
   MOZ_LOG(gUserInteractionPRLog, LogLevel::Debug, (msg, ##__VA_ARGS__))
 
-CanonicalBrowsingContext::CanonicalBrowsingContext(BrowsingContext* aParent,
+CanonicalBrowsingContext::CanonicalBrowsingContext(WindowContext* aParentWindow,
                                                    BrowsingContextGroup* aGroup,
                                                    uint64_t aBrowsingContextId,
                                                    uint64_t aOwnerProcessId,
                                                    uint64_t aEmbedderProcessId,
                                                    BrowsingContext::Type aType,
                                                    FieldTuple&& aFields)
-    : BrowsingContext(aParent, aGroup, aBrowsingContextId, aType,
+    : BrowsingContext(aParentWindow, aGroup, aBrowsingContextId, aType,
                       std::move(aFields)),
       mProcessId(aOwnerProcessId),
       mEmbedderProcessId(aEmbedderProcessId) {